components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java

```diff
@@ -107,28 +108,40 @@ public void create(IdentityRequest.IdentityRequestBuilder builder, HttpServletRe
                 log.debug("DCR request json : " + jsonData.toJSONString());
             }
 
-            RegistrationRequestProfile registrationRequestProfile =
-                    registerRequestBuilder.getRegistrationRequestProfile();
+            RegistrationRequestProfile registrationRequestProfile = registerRequestBuilder
+                    .getRegistrationRequestProfile();
             if (registrationRequestProfile == null) {
                 registrationRequestProfile = new RegistrationRequestProfile();
             }
 
             Object obj = jsonData.get(RegistrationRequest.RegisterRequestConstant.REDIRECT_URIS);
             if (obj != null && obj instanceof JSONArray) {
                 JSONArray redirectUris = (JSONArray) obj;
-                for (int i = 0; i < redirectUris.size(); i++) {
-                    registrationRequestProfile.getRedirectUris().add(redirectUris.get(i).toString());
+                for (Object redirectUri : redirectUris) {
+                    try {
+                        if (IdentityUtil.isNotBlank((String) redirectUri) && !redirectUri.equals("null")) {
```

Comment:
A null value can be casted to String. AFAIK this wont cause a ClassCastException